elasticapm/transport/base.py

```diff
@@ -135,7 +158,18 @@ def close(self):
         Cleans up resources and closes connection
         :return:
         """
-        self.flush(sync=True, start_flush_timer=False)
+        self._closed = True
+        self.queue("close", None)
+        if self._event_process_thread.is_alive():
+            self._event_process_thread.join()
+
+    def flush(self):
+        """
+        Trigger a flush of the queue.
+        Note: the flush happens asynchronously in a background thread, which means that the queue won't be immediately
```

Comment:
In the Go agent I make the Flush method block until the server has sent the queued events to the server, successfully or not. I did this primarily to make testing simpler and more robust. The way I did it is roughly this:

- when `flush` is called, check if the transport has a `flushed` event attribute already. If not, set a new one, and then send the flush request to the queue and wait on the event to be signalled
- when the server has finished processing the flush request, signal the event and clear the `flushed` event field

You might want to do this eventually, otherwise you'll probably run into intermittent failures on slow CI machines due to the sleeps.